cmd/node-termination-handler.go

```diff
@@ -313,6 +313,15 @@ func drainOrCordonIfNecessary(interruptionEventStore *interruptioneventstore.Sto
 	} else {
 		cordonAndDrainNode(node, nodeName, metrics, recorder, nthConfig.EnableSQSTerminationDraining)
 	}
+	podNameList, err := node.FetchPodNameList(nodeName)
```

Comment:
Looks like this will slightly change the behavior - previously if the node didn't exist then this code wouldn't execute, but now that you've moved this block out of the `else` block from line 349, this can get ran when the node is gone.

Perhaps we should return errors from the `cordonNode` and `cordonAndDrainNode` funcs and then check against them here?

Reply:
Do we want to show the list of of pods during drain? My understanding was that these pods are being drained anyway so why  care. I guess maybe they just want to maintain list of pods getting drained, is it?

Reply:
yeah, I don't really see a reason not to. Only users that want it have to put it into their webhook template.